labelbox_dev/task.py

```diff
@@ -221,3 +223,77 @@ def errors(self) -> Union[Dict[str, Any], None]:
             errors['failed_data_rows'] = task_result['failedDataRows']
 
         return errors if errors else None
+
+
+class UpdateDataRowsTask(BulkTask):
+    """ Asynchronous task representing bulk data_rows update job
+    """
+
+    task_type = TaskType.UPDATE_DATA_ROWS
+
+    def __init__(self, json):
+        super().__init__(json)
+
+    @lru_cache()
+    def fetch_result(self) -> Dict[str, Any]:
+        if self.status == TaskStatus.IN_PROGRESS.name:
+            raise ResourceNotFoundError(
+                f"Task result for task '{self.id}' not found. Task is still in progress"
+            )
+        else:
+            response = requests.get(self.result_url)
+            response.raise_for_status()
+            return response.json()
+
+    @property
+    def results(self) -> Union[Dict[str, Any], None]:
+        task_result = self.fetch_result()
+        if self.status == TaskStatus.FAILED.name:
+            logger.warning(
+                "Task has failed. Please look at `task.errors` for more details"
```

Comment:
+1, if keeping as logging, should be `logger.error`

Reply:
There could be a case where we partially succeed, in which case we still want Task.results() to return results for partially succeeded result. I'll keep the warning, but remove the `return None`, so that the `results()` can still return the partially succeeded data

Reply:
Since there are cases for both raising an exception and partial success, should we have a different TaskStatus for partial success? Another way to do it could be to have some "warnings" returned with SUCCESS status.